platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/TreasuryPapiExpansion.java

```diff
@@ -84,19 +78,27 @@ public boolean persist() {
 
     @Override
     public String onRequest(@Nullable OfflinePlayer player, @NotNull String param) {
-        for (TreasuryPapiHook hook : hooks) {
-            if (param.startsWith(hook.getPrefix())) {
-                // Pass request to specified hook sans prefix.
-                return hook.onRequest(player, param.replace(hook.getPrefix(), ""));
+        return base.onRequest(new PlayerData() {
+            @Override
+            public @Nullable String name() {
+                return player != null ? player.getName() : null;
             }
-        }
 
-        return null;
+            @Override
+            public @Nullable UUID uniqueId() {
+                return player != null ? player.getUniqueId() : null;
+            }
+
+            @Override
+            public @Nullable String getLocale() {
+                return player != null && player.isOnline() ? player.getPlayer().getLocale() : null;
```

Comment:
I would rather cache `player#getPlayer` into a variable because `OfflinePlayer#isOnline` is doing the same not null check whenever the player is online or not.
```suggestion
                if (player == null) {
                    return false;
                }

                Player bukkitPlayer = player.getPlayer();
                return bukkitPlayer != null ? bukkitPlayer.getLocale() : null;
```